### PR TITLE
New languages for translation and transliteration 

### LINKED ISSUE #781: Gurbani Text in Hindi
Are there any plans to include Gurbani Text in Hindi as one of the display options? Much of Sangat in India would love to have Hindi option too in parts outside Punjab like Delhi, Maharashtra, UP and various other places.
You might think it as a bad idea but as a beta release, we can use same text as Gurbani text but present it with a different Hindi font, and replace few characters that do not translate correctly. For e.g. translating ਰਜਾਈ from Gurmukhi will translate to रजाइी in Hindi using Hindi font, and hence will have to be corrected to रजाई by replacing इी with ई

### LINKED ISSUE #827: Add Shamukhi Translit

### LINKED ISSUE #828: Add Spanish Translations

### LINKED ISSUE #843: Add new settings section for "Language Options"
- Dropdown for Translation Language
- Dropdown for Transliteration Language 

This section should be right below "Fields to Display" section

### LINKED ISSUE #844: Rename "English Translation" setting to just "Translation"
In both the Font Slider settings and Fields to Display

### LINKED ISSUE #849: BaniOverlay should match which languages are chosen in the settings
For translations and transliteration

### LINKED ISSUE #850: Update Chromecast to support additional Translit and Translation languages

### PR DESCRIPTION
**Pull request by inderpreetsingh**
* [x] Adds support for Shahmukhi Transliteration [Fixes #827]
* [x] Adds Spanish Translations [Fixes #828]
* [x] Adds Devnagri [Fixes #781]
* [x] New Language Settings [ Fixes #843 and Fixes #844]
* [x] New languages to overlay [ Fixes #849]
* [x] Rename English Translations to Translations [Fixes #845 ]
* [x] SQLite support
* [x] Get anvaad-js to work 
* [x] Update overlay (and other places) to support anvaad-js transliterations

### CODE REVIEW

File: www/js/viewer.js [hunk 1]
```diff
@@ -298,21 +298,50 @@ const createCards = (rows, LineID) => {
     gurmukhiContainer.innerHTML = `<span class="padchhed">${padched}</span>
                                     <span class="larivaar">${larivaar}</span>`;
 
-    const englishContainer = document.createElement('div');
-    englishContainer.innerHTML = row.English || '';
+    const enTranslation = h('div.english-translation.transtext');
+    enTranslation.innerHTML = row.English || '';
+    const esTranslation = h('div.spanish-translation.transtext');
+    // If spanish translation is not available fall back to english
```
Comment (tsingh777) [inline www/js/viewer.js]: should we do this? cc @maneetpaul

File: www/js/viewer.js [hunk 2]
```diff
@@ -298,21 +298,50 @@ const createCards = (rows, LineID) => {
...
+    );
+    const dnTransliteration = h(
+      'div.devnagri-transliteration.translittext',
+      row.Transliteration.Devnagri || '',
+    );
+    const enTransliteration = h(
+      'div.english-transliteration.translittext',
+      row.Transliteration.English || '',
+    );
+    const transliterationsContainer = h(
```
Comment (tsingh777) [inline www/js/viewer.js]: this approach doesn't scale well IMO. I'm fine with it for now since it will work, but I'd like to revisit in the future.

File: app.js [hunk 3]
```diff
@@ -253,8 +253,13 @@ function createBroadcastFiles(arg) {
 
 const showLine = (line, socket = io) => {
   const overlayPrefs = store.get('obs');
-  const payload = Object.assign(line, overlayPrefs);
-  if (!line.fromScroll) {
+  const lineWithSettings = line;
```
Comment (tsingh777) [inline app.js]: do we want to include the new translation as a broadcast file?

Comment (tsingh777) [inline app.js]: Are chromecast changes not part of this PR?

File: www/obs/index.html [hunk 4]
```diff
@@ -12,7 +12,10 @@
         Gurmukhi: 'ik®pw kry jy AwpxI qw hir rKw aur Dwir ]1]',
         English: 'If the Lord shows His Mercy, then I keep Him enshrined within my heart. ||1||',
         Punjabi: 'jy prmwqmw AwpxI myhr kry, qW mYN BI aus dw nwm ihrdy ivc pro r`KW [1[',
-        Transliteration: 'kirapaa kare je aapanee taa har rakhaa aur dhaar ||1||',
+        Transliteration: {
+          en: 'kirapaa kare je aapanee taa har rakhaa aur dhaar ||1||',
+        },
+        Spanish: 'Cuando el Señor así nos bendice, enaltecemos al Guru en la mente. (1)',
```
Comment (tsingh777) [inline www/obs/index.html]: nitpicking - should the preview in the obs window have the correct languages?

File: www/js/banidb/sqlite-search.js [hunk 5]
```diff
@@ -329,7 +329,7 @@ const loadBani = (BaniID, BaniLength) =>
       init();
     }
     db.all(
-      `SELECT v.ID, v.Gurmukhi, v.Visraam, v.MainLetters, v.English, v.Transliteration,
+      `SELECT v.ID, v.Gurmukhi, v.Visraam, v.MainLetters, v.English, v.Transliteration, v.Translations
```
Comment (tsingh777) [inline www/js/banidb/sqlite-search.js]: improvement idea - include spanish translations of ceremony (the english bits)

File: www/js/viewer.js (hunk 1 above)
Comment (inderpreetsingh) [reply www/js/viewer.js]: I did this specially for the ceremonies where you have full page filled with english (explanations) because it would otherwise show nothing - so it goes to spanish if english is not available.

File: app.js (hunk 3 above)
Comment (inderpreetsingh) [reply app.js]: Yes, I will send a separate PR for that today.

File: www/js/viewer.js (hunk 2 above)
Comment (inderpreetsingh) [reply www/js/viewer.js]: What approach would you suggest?

Comment (tsingh777) [reply www/js/viewer.js]: react 😄

Comment (tsingh777) [reply www/js/viewer.js]: No need to change now.

File: www/js/viewer.js (hunk 1 above)
Comment (inderpreetsingh) [reply www/js/viewer.js]: Note: Make it always english.

**Comment by tsingh777** [comment]
see https://github.com/KhalisFoundation/sttm-chromecast/pull/23#pullrequestreview-329585797
